### PR TITLE
FIX: Paths and pyarrow

### PR DESCRIPTION
**Pull request by larsoner**
### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)

Some better error messages for path problems, and contains fix for failure in https://github.com/mne-tools/mne-bids-pipeline/pull/833

### CODE REVIEW

**Comment by larsoner** [comment]
@hoechenberger CIs fail with:
```
RuntimeError: Error 404 when trying to download /home/circleci/mne_data/ds003775/sub-010/ses-t1/sub-010_ses-t1_scans.tsv from https://s3.amazonaws.com/openneuro.org/ds003775/sub-010/ses-t1/sub-010_ses-t1_scans.tsv?versionId=hAC_TNr30jHiLxmLX7ergUVJILPtRjXo
```
Do we need to open an issue with the openneuro folks about this?

**Comment by hoechenberger** [comment]
I would guess so :/

**Comment by larsoner** [comment]
Okay we wait for https://github.com/OpenNeuroOrg/openneuro/issues/2976

File: mne_bids_pipeline/tests/datasets.py [hunk 1]
```diff
@@ -85,6 +85,8 @@ class DATASET_OPTIONS_T(TypedDict, total=False):
     "ds003775": {
         "openneuro": "ds003775",
         "include": ["sub-010"],
+        # See https://github.com/OpenNeuroOrg/openneuro/issues/2976
+        "exclude": ["sub-010/ses-t1/sub-010_ses-t1_scans.tsv"],
```
Comment (larsoner) [inline mne_bids_pipeline/tests/datasets.py:89]: No idea how long it'll take them to fix the issue so I'm going to see if we can just vendor the file in the meantime

**Comment by larsoner** [comment]
... and I trivially changed us to use `ruff-format` rather than `black` for formatting to follow MNE-Python

File: mne_bids_pipeline/tests/datasets.py (hunk 1 above)
Comment (larsoner) [reply mne_bids_pipeline/tests/datasets.py:89]: It seems to have worked!

**Pull request merged**